src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -1040,5 +1086,23 @@ private static void CheckTypeForwardedTo(Assembly sourceAssembly, Assembly destA
         {
             // nop on core
         }
+
+        internal sealed class TopLevelAssemblyTypeResolver
+        {
```

Comment:
I assume this came from the desktop given its retro style - I think a closure at the Type.GetType() callsite would be clearer these days.